### ISSUE TITLE
1.x selects 1.11beta1

### ISSUE THREAD

**Issue by simonpasquier** [travis-ci/gimme#152]
According to the README:

> Use of .x will not find release candidates, alphas, betas or other non-release versions: it's only for finding the last stable release

But `1.x` currently returns `v1.11beta1`

```
$ gimme --resolve 1.x
1.11beta1
$ gimme --resolve stable
1.10.3
```

**Comment by philpennock** [comment]
Confirmed.  The `.x` matching is designed for and was only tested for patch-level versions.  Looks like there's a bug when used in the minor position.

**Comment by philpennock** [comment]
(That is to say: this is a valid bug, we never say to not use it in the minor position, and it makes sense, so we should make that work)

**Comment by simonpasquier** [comment]
@philpennock thanks for the quick confirmation!

**Comment by philpennock** [comment]
I stopped for breakfast before handling the tests, but have now done that.  I think that the code in the PR should work for you.  Can you confirm?

**Comment by simonpasquier** [comment]
@philpennock yep, it's working for me. Eventually we're going to pin our TravisCI configuration to `1.10.x` as this is more deterministic.

**Comment by dmitshur** [comment]
Thanks for resolving this issue in `gimme`!

Is there an existing issue tracking the task of updating the version of `gimme` that travis-ci.org uses?

I tried restarting [my build](
https://travis-ci.org/shurcooL/home/jobs/402746128) just now, but it’s still using `gimme v1.5.0` where this bug is not yet resolved:

```
$ GIMME_OUTPUT="$(gimme 1.x | tee -a $HOME/.bashrc)" && eval "$GIMME_OUTPUT"
go version go1.11beta2 linux/amd64
$ gimme version
v1.5.0
```

**Comment by philpennock** [comment]
(Just to avoid radio silence: I'm not a Travis employee, I can't speak to Travis; I'm an external contributor to `gimme`.  So someone else would have to answer, although I suspect that an issue against the main Travis repo is more likely to get a response).

**Comment by dmitshur** [comment]
Thanks for the heads up @philpennock. I reported it to their support email, and afterwards found https://github.com/travis-ci/travis-ci/issues/9652, which looks like a decent issue for the purpose of tracking this issue.

I suspect `gimme` will need to release a newer version that includes this fix (the current latest is v1.5.0, which does not).
